### PR TITLE
PROD-6354 Added settings page for reactions

### PR DESCRIPTION
**Pull request by surajkrsingh**
### Jira Issue:
https://buddyboss.atlassian.net/browse/PROD-6354
https://buddyboss.atlassian.net/browse/PROD-6284

### General Note
Keep all conversations related to this PR in the associated Jira issue(s). Do NOT add comment on this PR or edit this PR’s description.

### Notes to Developer
* Ensure the IDs (i.e. PROD-1) of all associated Jira issues are reference in this PR’s title
* Ensure that you have achieved the Definition of Done before submitting for review
* When this PR is ready for review, move the associate Jira issue(s) to “Needs Review” (or “Code Review” for Dev Tasks)

### Notes to Reviewer
* Ensure that the Definition of Done have been achieved before approving a PR
* When this PR is approved, move the associated Jira issue(s) to “Needs QA” (or “Approved” for Dev Tasks)

### CODE REVIEW

**Review by KartikSuthar — DISMISSED** [review]

**Review by KartikSuthar — DISMISSED** [review]

**Review by KartikSuthar — DISMISSED** [review]

File: src/bp-core/admin/bp-core-admin-settings.php [hunk 1]
```diff
@@ -3479,3 +3464,152 @@ function bp_admin_setting_callback_custom_logout_redirection() {
...
+		),
+	);
+
+	$reactions_modes = apply_filters( 'bb_setting_reaction_mode_args', $reactions_modes );
+
+	if ( ! empty( $reactions_modes ) && is_array( $reactions_modes ) ) {
+		$notice_text = '';
+		foreach ( $reactions_modes as $reaction_mode ) {
+			?>
+			<label for="<?php echo $reaction_mode['id']; ?>" class="<?php echo esc_attr( ! empty( $reaction_mode['disabled'] ) ? 'disabled' : '' ); ?>">
```
Comment (chetansatasiya) [inline src/bp-core/admin/bp-core-admin-settings.php:3542]:
```suggestion
			<label for="<?php echo esc_attr( $reaction_mode['id'] ); ?>" class="<?php echo esc_attr( ! empty( $reaction_mode['disabled'] ) ? 'disabled' : '' ); ?>">
```

File: src/bp-core/admin/bp-core-admin-settings.php [hunk 2]
```diff
@@ -3479,3 +3464,152 @@ function bp_admin_setting_callback_custom_logout_redirection() {
...
+				<input name="<?php echo $reaction_mode['name']; ?>"
```
Comment (chetansatasiya) [inline src/bp-core/admin/bp-core-admin-settings.php:3543]: Same here add the escaping functions

File: src/bp-core/admin/bp-core-admin-settings.php [hunk 3]
```diff
@@ -3479,3 +3464,152 @@ function bp_admin_setting_callback_custom_logout_redirection() {
...
+					id="<?php echo $reaction_mode['id']; ?>"
+					type="radio"
+					value="<?php echo $reaction_mode['value']; ?>"
+					data-current-val="<?php echo bb_get_reaction_mode(); ?>"
```
Comment (chetansatasiya) [inline src/bp-core/admin/bp-core-admin-settings.php:3547]: Same here add the escaping functions

File: src/bp-core/admin/bp-core-admin-settings.php [hunk 4]
```diff
@@ -3479,3 +3464,152 @@ function bp_admin_setting_callback_custom_logout_redirection() {
...
+
+			if ( ! empty( $reaction_mode['is_checked'] ) ) {
+				$notice_text = $reaction_mode['notice'];
+			}
+		}
+
+		if ( ! empty( $notice_text ) ) {
+			?>
+			<p class="description bb-reaction-mode-description">
+				<?php echo $notice_text; ?>
```
Comment (chetansatasiya) [inline src/bp-core/admin/bp-core-admin-settings.php:3566]: Same here add the escaping functions

**Review by chetansatasiya — DISMISSED** [review]

**Pull request merged**